applications/fishing-map/src/features/workspace/workspace.selectors.ts

```diff
@@ -18,20 +34,78 @@ export const selectWorkspaceViewport = createSelector([selectWorkspace], (worksp
   return workspace?.viewport
 })
 
-export const selectWorkspaceDataviews = createSelector([selectWorkspace], (workspace) => {
-  if (!workspace) return
-  // TODO resolve them here
-  return workspace.dataviews
-})
+export const selectWorkspaceDataviewsResolved = createSelector(
```

Comment:
I'm a bit puzzled by the logic... Why not a much simpler workspaceDataviewsConfig.map and then get the corresponding dataview for each workspaceDataviewConfig?
My idea was that a workspaceDataviewConfig would always be needed, not optional. So workspace.dataviewsConfig should have a dataviewConfig for the background and the basemap, event though it only uses the dataviews defaults.

Reply:
Totally agree, I mess up things instead of keeping it simple, I've included the dataviewConfig for the basemap and now it is much easier to loop only thought `workspaceDataviewConfig` see [this commit](https://github.com/GlobalFishingWatch/frontend/pull/118/commits/5b98d0ee7bb9914d49f0ecdf0e2c70ead87b009b)